flang/lib/Lower/OpenACC.cpp

```diff
@@ -4187,21 +4187,24 @@ void Fortran::lower::attachDeclarePostDeallocAction(
 
   std::stringstream fctName;
   fctName << converter.mangleName(sym) << declarePostDeallocSuffix.str();
-  mlir::Operation &op = builder.getInsertionBlock()->back();
-  if (op.hasAttr(mlir::acc::getDeclareActionAttrName())) {
-    auto attr = op.getAttrOfType<mlir::acc::DeclareActionAttr>(
+  mlir::Operation *op = &builder.getInsertionBlock()->back();
+  if (mlir::isa<fir::ResultOp>(*op))
```

Comment:
nit: maybe it is worth adding an assertion here that `ResultOp` does not have operands - in this case, it may happen that the operand definition operation precedes `ResultOp`.  It seems it may go wrong other different ways, but asserting the suspicious case with operands in `ResultOp` looks reasonable to me.

Reply:
Sure I'll add it.